test/opt/set_spec_const_default_value_test.cpp

```diff
@@ -537,6 +541,519 @@ INSTANTIATE_TEST_CASE_P(
+            // default values
+            SpecIdToValueBitPatternMap{
+                {100, {0x0, 0x0, 0x0, 0x0}},
+                {101, {0x10101010}},
+                {102, {0x0, 0x0, 0x0, 0x2}},
+            },
+            // expected
+            "OpDecorate %1 SpecId 100\n"
+            "OpDecorate %2 SpecId 101\n"
+            "OpDecorate %3 SpecId 102\n"
+            "%bool = OpTypeBool\n"
+            "%1 = OpSpecConstantFalse %bool\n"
+            "%2 = OpSpecConstantTrue %bool\n"
+            "%3 = OpSpecConstantTrue %bool\n",
+        },
+    }));
+
+INSTANTIATE_TEST_CASE_P(
+    InvalidCases, SetSpecConstantDefaultValueInBitPatternFormParamTest,
+    ::testing::ValuesIn(std::vector<
+                        SetSpecConstantDefaultValueInBitPatternFormTestCase>{
+        // 0. Do not crash when decoration group is not used.
+        {
+            // code
+            "OpDecorate %1 SpecId 100\n"
+            "%1 = OpDecorationGroup\n"
+            "%int = OpTypeInt 32 1\n"
+            "%3 = OpSpecConstant %int 100\n",
+            // default values
+            SpecIdToValueBitPatternMap{{100, {0x7fffffff}}},
+            // expected
+            "OpDecorate %1 SpecId 100\n"
+            "%1 = OpDecorationGroup\n"
+            "%int = OpTypeInt 32 1\n"
+            "%3 = OpSpecConstant %int 100\n",
+        },
+        // 1. Do not crash when target does not exist.
+        {
+            // code
+            "OpDecorate %1 SpecId 100\n"
+            "%int = OpTypeInt 32 1\n",
+            // default values
+            SpecIdToValueBitPatternMap{{100, {0x7fffffff}}},
+            // expected
+            "OpDecorate %1 SpecId 100\n"
+            "%int = OpTypeInt 32 1\n",
+        },
+        // 2. Do nothing when SpecId decoration is not attached to a
+        // non-spec-contant instruction.
+        {
+            // code
+            "OpDecorate %1 SpecId 100\n"
+            "%int = OpTypeInt 32 1\n"
+            "%int_101 = OpConstant %int 101\n",
+            // default values
+            SpecIdToValueBitPatternMap{{100, {0x7fffffff}}},
+            // expected
+            "OpDecorate %1 SpecId 100\n"
+            "%int = OpTypeInt 32 1\n"
+            "%int_101 = OpConstant %int 101\n",
+        },
+        // 3. Do nothing when SpecId decoration is not attached to a
+        // OpSpecConstant{|True|False} instruction.
+        {
+            // code
+            "OpDecorate %1 SpecId 100\n"
+            "%int = OpTypeInt 32 1\n"
+            "%3 = OpSpecConstant %int 101\n"
+            "%1 = OpSpecConstantOp %int IAdd %3 %3\n",
+            // default values
+            SpecIdToValueBitPatternMap{{100, {0x7fffffff}}},
+            // expected
+            "OpDecorate %1 SpecId 100\n"
+            "%int = OpTypeInt 32 1\n"
+            "%3 = OpSpecConstant %int 101\n"
+            "%1 = OpSpecConstantOp %int IAdd %3 %3\n",
+        },
+        // 4. Do not crash and do nothing when SpecId decoration is applied to
```

Comment:
I'm glad this is a no-op.
I suspect that's an invalid scenario, but it probably falls into a gray area.